tests/pr_test.go

```diff
@@ -11,35 +17,180 @@ import (
 const resourceGroup = "geretain-test-cbr"
 const zoneExampleTerraformDir = "examples/zone"
 const completeExampleTerraformDir = "examples/multizone-rule"
+const permanentResourcesYaml = "../common-dev-assets/common-go-assets/common-permanent-resources.yaml"
 
 func TestRunZoneExample(t *testing.T) {
 	t.Parallel()
+	cloudInfoSvc, err := cloudinfo.NewCloudInfoServiceFromEnv("TF_VAR_ibmcloud_api_key", cloudinfo.CloudInfoServiceOptions{})
+	assert.Nil(t, err, "Failed to create cloud info service")
 
 	options := testhelper.TestOptionsDefaultWithVars(&testhelper.TestOptions{
 		Testing:       t,
 		TerraformDir:  zoneExampleTerraformDir,
 		Prefix:        "cbr-zone",
 		ResourceGroup: resourceGroup,
 	})
-
+	options.SkipTestTearDown = true
 	output, err := options.RunTestConsistency()
 	assert.Nil(t, err, "This should not have errored")
 	assert.NotNil(t, output, "Expected some output")
+
+	outputs := terraform.OutputAll(options.Testing, options.TerraformOptions)
+
+	zone, err := cloudInfoSvc.GetCBRZoneByID(outputs["zone_id"].(string))
+	assert.Nilf(t, err, "This should not have errored, could not get zone")
+
+	expectedAddresses := []contextbasedrestrictionsv1.AddressIntf{
+		&contextbasedrestrictionsv1.AddressVPC{
+			Type:  core.StringPtr("vpc"),
+			Value: core.StringPtr(outputs["vpc_crn"].(string)),
+		},
+		&contextbasedrestrictionsv1.AddressServiceRef{
+			Type: core.StringPtr("serviceRef"),
+			Ref: &contextbasedrestrictionsv1.ServiceRefValue{
+				AccountID:   core.StringPtr(outputs["account_id"].(string)),
+				ServiceName: core.StringPtr("secrets-manager"),
+			},
+		},
+	}
+	assert.Equal(t, outputs["zone_name"].(string), *zone.Name)
+	assert.Equal(t, outputs["zone_description"].(string), *zone.Description)
+	assert.Equal(t, outputs["account_id"].(string), *zone.AccountID)
+	assert.EqualValues(t, expectedAddresses, zone.Addresses)
+	assert.Empty(t, zone.Excluded)
+
+	options.TestTearDown()
 }
 
 func TestRunCompleteExample(t *testing.T) {
 	t.Parallel()
+	cloudInfoSvc, err := cloudinfo.NewCloudInfoServiceFromEnv("TF_VAR_ibmcloud_api_key", cloudinfo.CloudInfoServiceOptions{})
+	assert.Nil(t, err, "Failed to create cloud info service")
 
-	options := testhelper.TestOptionsDefaultWithVars(&testhelper.TestOptions{
-		Testing:       t,
-		TerraformDir:  completeExampleTerraformDir,
-		Prefix:        "cbr-multizone",
-		ResourceGroup: resourceGroup,
-	})
+	permanentResources, err := common.LoadMapFromYaml(permanentResourcesYaml)
```

Comment:
Just a note: Some developers have been loading the permanent resources from yaml in a `TestMain()` and assigned to test package scope variables, so that they are loaded once and available to all tests. In this case you only need it once, just mentioning it in case you might need these values in more than one test.